ddcommon/src/container_id.rs

```diff
@@ -85,10 +89,27 @@ fn extract_container_id(filepath: &Path) -> Result<String, Box<dyn std::error::E
     Err(ContainerIdNotFoundError.into())
 }
 
+/// # Safety
+/// Must not be called in multi-threaded contexts
+pub unsafe fn set_cgroup_file(file: String) {
+    TESTING_CGROUP_PATH = Some(file)
+}
+
+fn get_cgroup_path() -> PathBuf {
+    // Safety: we assume set_cgroup_file is not called when it shouldn't
+    if let Some(path) = unsafe { TESTING_CGROUP_PATH.as_ref() } {
+        Path::new(path.as_str()).into()
+    } else {
+        Path::new(DEFAULT_CGROUP_PATH).into()
+    }
+}
+
 pub fn get_container_id() -> Option<&'static str> {
     // cache container id in a static to avoid recomputing it at each call
+
     lazy_static! {
```

Comment:
noob question: lazy_static! guarantees thread safety ?

Reply:
Yes IIRC it uses CellOnce underneath to make this threadsafe.